src/main/java/network/brightspots/rcv/TabulatorSession.java

```diff
@@ -131,6 +131,10 @@ void convertToCdf() {
   }
 
   void tabulate() {
+    tabulate(null);
+  }
+
+  void tabulate(List<String> exceptionsEncountered) {
```

Comment:
Why not just have this return the List?  That's a more typical api for error codes and then you don't need to do all the checks for if (exceptionsEncountered != null) here.

Reply:
You can also get rid of the wrapper methods

Reply:
I'm OK with that approach also. I was thinking it would be weird, but upon further reflection, I guess it's fine.